api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -471,8 +488,26 @@ public static int compareVersion(String version, String value) {
 	 * @return true if version contains qualifier
 	 */
 	public static boolean isVersionWithQualifier(String version) {
-		Matcher matcher = Pattern.compile("(\\d+)\\.(\\d+)(\\.(\\d+))?(\\-([A-Za-z]+))").matcher(version);
-		return matcher.matches();
+		Matcher matcher = VERSION_PATTERN.matcher(version);
+		return matcher.matches() && !"".equals(matcher.group("snapshot"));
+		Matcher versionMatcher = VERSION_PATTERN.matcher(version);
+		Matcher valueMatcher = VERSION_PATTERN.matcher(value);
+
+		if (!versionMatcher.matches() || !valueMatcher.matches()) {
+			// this seems wrong
+			return 0;
+		}
+
+		final String[] versions = {
+			versionMatcher.group("major"),
+			versionMatcher.group("minor"),
+			versionMatcher.group("patch")
+		};
+		final String[] values = {
+			valueMatcher.group("major"),
+			valueMatcher.group("minor"),
+			valueMatcher.group("patch")
+		};
```

Comment:
Sorry, this block was intended to replace lines 426-450 above (the part in `compareVersion()` that is breaking the string down into it's component parts). The loop from lines 452-454 will also need to be adjusted.